docs/proposals/csi.md

```diff
@@ -0,0 +1,359 @@
+  * [Goals](#goals)
+  * [Non\-goals](#non-goals)
+  * [Proposal](#proposal)
+    * [Requirement](#requirement)
+    * [Architecture](#architecture)
+    * [Workflow](#workflow)
+    * [Deployment](#deployment)
+    * [Example](#example)
+  * [Work Items](#work-items)
+
+## Motivation
+
+Currently KubeEdge only supports the following volumes which are based on Kubernetes.
+
+* [emptyDir](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)
+* [hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)
+* [configMap](https://kubernetes.io/docs/concepts/storage/volumes/#configmap)
+* [secret](https://kubernetes.io/docs/concepts/storage/volumes/#secret)
+
+That is not enough for the customers who are using KubeEdge.
+The customers often need to store data and run stateful applications at edge.
+For example, in a system which collects video data and then makes analysis,
+one application stores the video data into a shared storage,
+and another application reads the data from the storage to make analysis.
+KubeEdge should allow customers to store data using StorageClass (SC),
+PersistentVolume (PV) and PersistentVolumeClaim (PVC)
+so that the customers can deploy stateful applications at edge.
+
+The Container Storage Interface (CSI) is a specification that resulted from cooperation
+between community members including Kubernetes, Mesos, Cloud Foundry, and Docker.
+The goal of this interface is to establish a standardized mechanism
+to expose arbitrary storage systems to their containerized workloads.
+CSI Spec has already been released v1.1 and lots of CSI Drivers are released by vendors.
+CSI Volume Plugin is supported beginning with kubernetes v1.9 and in GA status since v1.13.
+
+* [CSI Spec](https://github.com/container-storage-interface/spec/blob/master/spec.md)
+* [CSI Drivers](https://kubernetes-csi.github.io/docs/drivers.html)
+* [CSI Volume Plugin in Kubernetes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md)
+
+We can support Container Storage Interface (CSI) in KubeEdge,
+and give a standardized solution to the customers who want to use persistent storage at edge.
+The customers can choose any kind of CSI Drivers on demands. Since KubeEdge is built upon Kubernetes,
+it is possible to reuse some of existing framework from Kubernetes in KubeEdge.
+
+## Goals
+
+* To support [basic CSI Volume Lifecycle](https://github.com/container-storage-interface/spec/blob/master/spec.md#volume-lifecycle) in KubeEdge.
+   * Create Volume and Delete Volume.
+   * Controller Publish Volume and Controller Unpublish Volume.
+   * Node Stage Volume and Node Unstage Volume.
+   * Node Publish Volume and Node Unpublish Volume.
+* Compatiable with Kubernetes and CSI.
+
+## Non-goals
+
+* To support Raw Block Volume.
+* To support Volume Snapshot and Restore.
+* To support Volume Topology.
+
+## Proposal
+
+### Requirement
+* Kubernetes v1.13+
```

Comment:
Just a Question, The Kubernetes master components version should be v1.13+ or the k8s.io/kubernetes in vendor should be 1.13+ or both ?

Reply:
@sids-b good question, It would be `both`
Since the CSI is GA after v1.13+.
https://kubernetes-csi.github.io/docs/introduction.html#kubernetes-releases